proxy/proxy.go

```diff
@@ -1231,6 +1232,7 @@ func (p *Proxy) serveResponse(ctx *context) {
 		p.tracing.setTag(ctx.proxySpan, ErrorTag, true)
 		p.tracing.setTag(ctx.proxySpan, StreamBodyEvent, StreamBodyError)
 		p.tracing.logStreamEvent(ctx.proxySpan, StreamBodyEvent, fmt.Sprintf("Failed to stream response: %v", err))
+		ctx.responseWriter.Flush()
```

Comment:
Why not flush within `copyStream` itself?

Reply:
@AlexanderYastrebov copyStream already flushes but it is not used for error conditions.

Reply:
Yes, and I propose to flush there right before returning the error

Reply:
I changed it to the defer in ServeHTTP, which seems to be most safe case for both cases.
The problem is that we have it also in sendError which is a different call path than streaming the body.

Reply:
as discussed internally, use the one in defer of ServeHTTP